src/states/bootstrapping_peer.rs

```diff
@@ -75,25 +109,62 @@ impl BootstrappingPeer {
             self.timer,
             self.peer_map,
             conn_infos,
+            relocate_payload,
         )))
     }
 
     fn send_bootstrap_request(&mut self, dst: ConnectionInfo) {
-        debug!("{} Sending BootstrapRequest to {}.", self, dst.peer_addr);
+        let _ = self.nodes_to_await.remove(&dst.peer_addr);
+
+        if self.bootstrap_connection.is_some() {
+            // we already have an active connection, drop this one
+            self.network_service
+                .service_mut()
+                .disconnect_from(dst.peer_addr);
+        } else {
+            debug!("{} Sending BootstrapRequest to {}.", self, dst.peer_addr);
 
-        let token = self.timer.schedule(BOOTSTRAP_TIMEOUT);
-        self.bootstrap_connection = Some((dst.clone(), token));
+            let token = self.timer.schedule(BOOTSTRAP_TIMEOUT);
+            self.bootstrap_connection = Some((dst.clone(), token));
 
-        let message =
-            if let Ok(message) = self.to_signed_direct_message(DirectMessage::BootstrapRequest) {
-                message
+            // If we are relocating, request bootstrap to the section matching the name given to us
+            // by our section. Otherwise request bootstrap to the section matching our current name.
+            let destination = if let Some(details) = self.relocate_details.as_ref() {
+                details.content().destination
             } else {
-                return;
+                *self.name()
             };
 
-        let conn_infos = vec![dst];
-        let dg_size = 1;
-        self.send_message_to_initial_targets(conn_infos, dg_size, message);
+            self.send_direct_message(&dst, DirectMessage::BootstrapRequest(destination));
+            self.peer_map_mut().connect(dst);
+        }
+    }
+
+    fn join_section(
+        &mut self,
+        prefix: Prefix<XorName>,
+        conn_infos: Vec<ConnectionInfo>,
+    ) -> Result<Transition, RoutingError> {
+        let old_full_id = self.full_id.clone();
+
+        if !prefix.matches(self.name()) {
+            self.full_id = FullId::within_range(&prefix.range_inclusive());
```

Comment:
Just to confirm, so here is the place that we are `switching` id?
and we will `forgot` the old id immediately after this function, right?

Reply:
That's right. I'm going to put a log output there because the id switch might be important when debugging.